### PR TITLE
Added class to perform univariate filtering based on inter quantile range

### PR DESCRIPTION
**Pull request by astro313**
## Description
New feature in preprocessing data -- performing filtering on a list of columns based on their interquartile range. 

## Checklist

- [x] I squashed my commits to a reasonable number of descriptive commits.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: primrose/transformers/filter.py [hunk 1]
```diff
@@ -94,3 +96,99 @@ def transform(self, data):
...
+
+
+    def fit(self, data):
+        """fit data, here just passing
+
+        Args:
+            data (object): some data
+
+        """
+        pass
```
Comment (reka-ww) [inline primrose/transformers/filter.py:127]: How do you make sure that we don't consider the testing data to determine IQR?

Comment (astro313) [reply primrose/transformers/filter.py:127]: @reka-ww  I'll fix that today.

File: primrose/transformers/filter.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-"""Transform data by filtering in data using filtering operations 
+"""Transform data by filtering in data using filtering operations
 
 Author(s):
     Michael Skarlinski (michael.skarlinski@weightwatchers.com)
```
Comment (briangrahamww) [inline primrose/transformers/filter.py:4]: @astro313 , you can add yourself to this list 😄

File: primrose/transformers/filter.py [hunk 3]
```diff
@@ -94,3 +96,99 @@ def transform(self, data):
...
+            Remove outlier, defined as 2.0x interquartile range (IQR) of a single variable.
+
+        Parameters
+        ----------
+        xvarlist: list of str
+            colnames to perform filtering on
+
+        """
+
+        df = df.dropna(inplace=False)
```
Comment (briangrahamww) [inline primrose/transformers/filter.py:142]: I think we have other transformers that have options for dealing with nans. We might want to keep this to the single use case of just filtering out outliers based on IQR.  That way it's explicitly declared with the rest of the data transformations if you are dropping nans and at what point in the pipeline.

**Review by briangrahamww — CHANGES_REQUESTED** [review]
I think it's best to separate the nan removal.

**Comment by kzecchini** [comment]
@astro313 @briangrahamww curious about the progress on this

**Comment by astro313** [comment]
After discussing with @reka-ww , we want to think more about how to implement this/should we implement this. Will close this PR for now.

**Pull request closed without merging**